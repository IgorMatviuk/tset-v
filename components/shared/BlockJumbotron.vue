<template>
  <div class="block-jumbotron">
    <BContainer class="block-jumbotron__container">
      <picture>
        <source
          :data-srcset="content.images.background_xl | webp"
          srcset="/img/stub.png"
          type="image/webp"
          media="(min-width: 1500px)"
        />
        <source
          :data-srcset="content.images.background_xl"
          srcset="/img/stub.png"
          type="image/jpeg"
          media="(min-width: 1500px)"
        />
        <source
          :data-srcset="content.images.background_lg | webp"
          srcset="/img/stub.png"
          type="image/webp"
          media="(min-width: 1200px)"
        />
        <source
          :data-srcset="content.images.background_lg"
          srcset="/img/stub.png"
          type="image/jpeg"
          media="(min-width: 1200px)"
        />
        <source
          :data-srcset="content.images.background_md | webp"
          srcset="/img/stub.png"
          type="image/webp"
          media="(min-width: 992px)"
        />
        <source
          :data-srcset="content.images.background_md"
          srcset="/img/stub.png"
          type="image/jpeg"
          media="(min-width: 992px)"
        />
        <source
          :data-srcset="content.images.background_sm | webp"
          srcset="/img/stub.png"
          type="image/webp"
          media="(min-width: 768px)"
        />
        <source
          :data-srcset="content.images.background_sm"
          srcset="/img/stub.png"
          type="image/jpeg"
          media="(min-width: 768px)"
        />
        <source
          :data-srcset="content.images.background_xs | webp"
          srcset="/img/stub.png"
          type="image/webp"
        />
        <img
          :data-srcset="content.images.background_xs"
          src="/img/stub.png"
          :alt="content.title"
          class="block-jumbotron__image lazyload"
          loading="lazy"
          width="100"
          height="100"
          data-object-fit="cover"
          data-object-position="center bottom"
        />
      </picture>
      <h1 class="block-jumbotron__title">
        <slot></slot>
      </h1>
    </BContainer>
  </div>
</template>

<script>
import imageNextGen from '@/mixins/image-next-gen'

export default {
  mixins: [imageNextGen],
  props: {
    content: {
      type: Object,
      required: true
    }
  }
}
</script>

<style lang="scss">
.block-jumbotron {
  position: relative;
  height: 340px;

  .lazyload,
  .lazyloading {
    opacity: 0;
  }

  .lazyloaded {
    opacity: 1;
    transition: opacity 0.3s;
  }

  &__container {
    display: flex;
    align-items: center;
    height: 100%;
  }

  &__image {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: -1;
    width: 100%;
    height: 100%;
    object-fit: cover;
    object-position: center;
  }

  &__title {
    margin-bottom: 47px;
    font-size: 30px;
    font-weight: 700;
    line-height: 35px;
    text-transform: uppercase;
  }

  @include media-breakpoint-up(md) {
    height: 396px;

    &__title {
      margin-bottom: 0;
      line-height: 40px;
    }
  }

  @include media-breakpoint-up(lg) {
    height: 455px;

    &__title {
      margin-left: 79px;
    }
  }

  @include media-breakpoint-up(xl) {
    height: 540px;

    &__title {
      margin-bottom: 32px;
      margin-left: 0;
      font-size: 40px;
      line-height: 50px;
    }
  }

  @include media-breakpoint-up(xxl) {
    &__title {
      margin-bottom: 61px;
      font-size: 50px;
      line-height: 60px;
    }
  }
}
</style>
