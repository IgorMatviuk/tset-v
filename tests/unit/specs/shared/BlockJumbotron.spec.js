import { mount, createLocalVue } from '@vue/test-utils'

import { BContainer } from 'bootstrap-vue'

import BlockJumbotron from '@/components/shared/BlockJumbotron'

const localVue = createLocalVue()
localVue.component('BContainer', BContainer)

describe('BlockPostNavigation', () => {
  let jumbotronContent

  function createBlockJumbotron() {
    const wrapper = mount(BlockJumbotron, {
      localVue,
      propsData: {
        content: {
          title: 'title',
          images: {
            background_xl: '/img/blog/background_xl.jpg',
            background_lg: '/img/blog/background_lg.jpg',
            background_md: '/img/blog/background_md.jpg',
            background_sm: '/img/blog/background_sm.jpg',
            background_xs: '/img/blog/background_xs.jpg'
          }
        }
      }
    })

    return wrapper
  }

  beforeEach(() => {
    jumbotronContent
  })

  test('render block jumbotron', () => {
    const wrapper = createBlockJumbotron()

    expect(wrapper.element).toMatchSnapshot()
  })
})
