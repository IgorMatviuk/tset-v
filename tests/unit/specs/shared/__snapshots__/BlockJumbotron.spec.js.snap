// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockPostNavigation render block jumbotron 1`] = `
<div
  class="block-jumbotron"
>
  <div
    class="block-jumbotron__container container"
  >
    <picture>
      <source
        data-srcset="/img/blog/background_xl.webp"
        media="(min-width: 1500px)"
        srcset="/img/stub.png"
        type="image/webp"
      />
       
      <source
        data-srcset="/img/blog/background_xl.jpg"
        media="(min-width: 1500px)"
        srcset="/img/stub.png"
        type="image/jpeg"
      />
       
      <source
        data-srcset="/img/blog/background_lg.webp"
        media="(min-width: 1200px)"
        srcset="/img/stub.png"
        type="image/webp"
      />
       
      <source
        data-srcset="/img/blog/background_lg.jpg"
        media="(min-width: 1200px)"
        srcset="/img/stub.png"
        type="image/jpeg"
      />
       
      <source
        data-srcset="/img/blog/background_md.webp"
        media="(min-width: 992px)"
        srcset="/img/stub.png"
        type="image/webp"
      />
       
      <source
        data-srcset="/img/blog/background_md.jpg"
        media="(min-width: 992px)"
        srcset="/img/stub.png"
        type="image/jpeg"
      />
       
      <source
        data-srcset="/img/blog/background_sm.webp"
        media="(min-width: 768px)"
        srcset="/img/stub.png"
        type="image/webp"
      />
       
      <source
        data-srcset="/img/blog/background_sm.jpg"
        media="(min-width: 768px)"
        srcset="/img/stub.png"
        type="image/jpeg"
      />
       
      <source
        data-srcset="/img/blog/background_xs.webp"
        srcset="/img/stub.png"
        type="image/webp"
      />
       
      <img
        alt="title"
        class="block-jumbotron__image lazyload"
        data-object-fit="cover"
        data-object-position="center bottom"
        data-srcset="/img/blog/background_xs.jpg"
        height="100"
        loading="lazy"
        src="/img/stub.png"
        width="100"
      />
    </picture>
     
    <h1
      class="block-jumbotron__title"
    />
  </div>
</div>
`;
